orca-webhook/src/main/groovy/com/netflix/spinnaker/orca/webhook/tasks/CreateWebhookTask.groovy

```diff
@@ -119,23 +124,13 @@ class CreateWebhookTask implements RetryableTask {
 
     if (statusCode.is2xxSuccessful() || statusCode.is3xxRedirection()) {
       if (stageData.waitForCompletion) {
-        def statusUrl = null
-        def statusUrlResolution = stageData.statusUrlResolution
-        switch (statusUrlResolution) {
-          case WebhookProperties.StatusUrlResolution.getMethod:
-            statusUrl = stageData.url
-            break
-          case WebhookProperties.StatusUrlResolution.locationHeader:
-            statusUrl = response.headers.getFirst(HttpHeaders.LOCATION)
-            break
-          case WebhookProperties.StatusUrlResolution.webhookResponse:
-            try {
-              statusUrl = JsonPath.compile(stageData.statusUrlJsonPath as String).read(response.body)
-            } catch (PathNotFoundException e) {
-              outputs.webhook << [error: e.message]
-              return TaskResult.builder(ExecutionStatus.TERMINAL).context(outputs).build()
-            }
+        String statusUrl
+        try {
+          statusUrl = determineWebHookStatusCheckUrl(response, stageData)
+        } catch (Exception e) {
+          log.error('Exception received while determining status change url', e)
```

Comment:
We no longer bubble up the json parsing error (or other errors from within `determineWebHookStatusCheckUrl`) to the user as we used to here:
https://github.com/spinnaker/orca/pull/3187/files#diff-e737f2c3e6b9c31027d9d0670960c694L135

this means when something goes wrong we will have to dig in the logs, but it's better if the user can find the exceptions in the execution and self-service fix it (hopefully)

Reply:
Good catch!.  Added the exception message to webhook output map.